tests/model_registry/conftest.py

```diff
@@ -289,14 +293,38 @@ def registered_model(request: FixtureRequest, model_registry_client: ModelRegist
 
 @pytest.fixture()
 def model_registry_operator_pod(admin_client: DynamicClient) -> Pod:
-    model_registry_operator_pods = [
-        pod
-        for pod in Pod.get(dyn_client=admin_client, namespace=py_config["applications_namespace"])
-        if re.match(MR_OPERATOR_NAME, pod.name)
-    ]
-    if not model_registry_operator_pods:
+    """Get the model registry operator pod."""
+    try:
+        for pod_list in TimeoutSampler(
+            wait_timeout=60,
+            sleep=5,
+            func=Pod.get,
+            dyn_client=admin_client,
+            namespace=py_config["applications_namespace"],
+        ):
+            for pod in pod_list:
+                if re.match(MR_OPERATOR_NAME, pod.name):
+                    return pod
+    except TimeoutExpiredError:
```

Comment:
You are returning the 1st match. Don't you want to wait for the number of matches to be 1? You can also potentially reuse get_pods_by_labels() here.